### ISSUE TITLE
Let labels take a list of custom colors

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#716]
## 🚀 Feature
Following up on discussion with @adamltyson on this [image.sc post](https://forum.image.sc/t/specify-labels-colors/31534) I think it would be nice if we allowed users to set their own list of colors for `add_labels` as an alternative to using random colors. If the user has more labels than colors we can just cycle through them.

**Comment by sofroniewn** [comment]
There is another request for this here https://github.com/hci-unihd/covid-if-annotations/issues/5

**Comment by constantinpape** [comment]
As mentioned in the other issue, I would have two use cases for this:
1. adding a mode where some labels in the segmentation layer are hidden, i.e. their alpha value is set to 0
2. Synchronize the colors between a label layer and a points layer.

I also looked at the image.sc post linked in the issue and I realized that for the second use-case the image layer with fixed color map makes more sense (I have just used the labels layer there to make zero transparent, otherwise I don't need any functionality provided by it).

However, for the first issue I think that passing a **dict** mapping label ids to colors would be the best solution. The color of a given label should then be determined like this:
```python
color = color_dict.get(label_id, random_color(label_id))
```
Also, I would need to turn this mode on and off, so it should be possible to set the `color_dict` dynamically:
```python
# set custom colors for the first labels
labels_layer.color_dict = {1: 'red', 2: 'green', 3: 'blue'}

# deactivate the custom colors
labels_layer.color_dict = {}
```

Let me know if this makes sense to you and if it's something that could be implemented in a rather short timeframe ;).

**Comment by sofroniewn** [comment]
> Let me know if this makes sense to you and if it's something that could be implemented in a rather short timeframe ;).

Yes, we should be able to turn this around pretty quickly (I'm imagining an initial PR in a few days from now, reviewed and merged by the end of the week). The more crisply we can specify it the faster.

I'm thinking that we will want to reuse some of the API paradigm that we use in the [points layer](https://napari.org/docs/api/napari.layers.points.html#napari.layers.points.points.Points) and consider adding support for labels `properties` (see #1214) either in or immediately after this PR.

The points layer currently has the following for face and edge color
```python
layer.*_colormap # colormap name, vispy colormap tuple
layer.*_color_mode # str with different modes, DIRECT, CYCLE, COLORMAP
layer.*_color_cycle # list of colors for the cycle
layer.*_color # rgba array, that could also be set to the name of a property
layer.properties # properties dict
```

For labels we could try and preserve this pattern, or instead of using a `CYCLE` use a `dict`. If we ignored the concept of `properties` we could just have

```python
layer.colormap # colormap name, vispy colormap tuple
layer.color_mode # str with different modes, DICT, COLORMAP
layer.color_dict # dict of colors {1: 'red', 2: 'green', 3: 'blue'}
```
The above would be a simple PR that could get you what you need.

If we do want to use properties we'd need to decide what they look like, and so I will continue some discussion of that on #1214. Curious what @kevinyamauchi and @jni make of this so far

**Comment by sofroniewn** [comment]
This is now closed by #1339
